### PR TITLE
feat: vault view

### PR DESCRIPTION
**Pull request by ChrisAmora**
This pull request introduces a new `Vault` view for the deployment system, providing a consolidated state summary of vault-related data across chains. It also improves error handling and test coverage for timelock balances, ensuring the system gracefully handles missing contracts and empty states.

**New Vault View Implementation:**

* Added `deployment/vault/view/view.go` to implement the `VaultView` structure, aggregating timelock balances, whitelisted addresses, and MCMS-with-timelock state across chains. This provides a single marshaled view for inspection and auditing.

**Testing Enhancements:**

* Added comprehensive tests in `deployment/vault/view/view_test.go` to verify `VaultView` generation under various scenarios, including environments with no chains, missing timelocks, and populated MCMS/whitelist data.

https://smartcontract-it.atlassian.net/browse/DX-1028

### CODE REVIEW

File: deployment/vault/view/view.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	WhitelistedAddresses map[uint64][]changeset.WhitelistEntry       `json:"whitelisted_addresses"`
+	MCMSWithTimelock     map[uint64]v1_0.MCMSWithTimelockView        `json:"mcms_with_timelock,omitempty"`
+}
+
+func (v *VaultView) MarshalJSON() ([]byte, error) {
+	type Alias VaultView
+	return json.MarshalIndent((*Alias)(v), "", "  ")
+}
+
+func VaultV2(e cldf.Environment, previousView json.Marshaler) (json.Marshaler, error) {
```
Comment (bytesizedroll) [inline deployment/vault/view/view.go]: Is `previousView` needed here?

**Review by ecPablo — DISMISSED** [review]

File: deployment/vault/view/view.go (hunk 1 above)
Comment (ChrisAmora) [reply deployment/vault/view/view.go]: I think it is required? https://github.com/smartcontractkit/chainlink-deployments-framework/blob/main/deployment/changeset.go#L128
Should I use ViewV1?
Added an `_` anyway.

**Comment by graham-chainlink** [comment]
View is legacy, are you sure you need it ? 

we have a warning at 
https://docs.cld.cldev.sh/guides/state-management/views
> Views are not intended to be supported long-term and will be replaced by CHIP ingestion.
You should only use Views if you require RDD (Requirement Definition Document) backwards compatibility.

**Comment by ChrisAmora** [comment]
> View is legacy, are you sure you need it ?
> 
> we have a warning at https://docs.cld.cldev.sh/guides/state-management/views
> 
> > Views are not intended to be supported long-term and will be replaced by CHIP ingestion.
> > You should only use Views if you require RDD (Requirement Definition Document) backwards compatibility.

I need an easy way for users to get offchain + onchain data. Is there an alternative?
If possible, can you add deprecation comments? Like this https://github.com/smartcontractkit/chainlink-deployments-framework/blob/dcc3dc14e40cbf19f76a1c74ffdfd13ba1077610/deployment/changeset.go#L21

**Review by kalverra — APPROVED** [review]

**Review by skudasov — APPROVED** [review]
